misc/docerina/src/main/java/org/ballerinalang/docgen/docs/BallerinaDocGenerator.java

```diff
@@ -195,6 +215,22 @@ public static void generateAPIDocs(Project project, String output, boolean exclu
         moduleLib.modules = getDocsGenModel(moduleDocMap, project.currentPackage().packageOrg().toString(),
                 project.currentPackage().packageVersion().toString());
         writeAPIDocs(moduleLib, Path.of(output), false, excludeUI);
+        copyIcon(project, Path.of(output), moduleLib);
+    }
+
+    public static void copyIcon(Project project, Path output, ModuleLibrary moduleLib) {
+        String sourceLocation = project.currentPackage().manifest().icon();
+        if (!sourceLocation.isEmpty()) {
+            output = output.resolve(moduleLib.modules.get(0).orgName).resolve(moduleLib.modules.get(0).id)
+                    .resolve(moduleLib.modules.get(0).version).resolve("icon.png");
+            Path iconPath = Paths.get(sourceLocation);
+            try {
+                var iconByteArray = Files.readAllBytes(iconPath);
```

Comment:
Shall we use strict typing here?
```suggestion
                byte[] iconByteArray = Files.readAllBytes(iconPath);
```

Reply:
Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/42386/commits/fff5836b373e5e37d3680640581edf928a6344c0